qemu/tests/cfg/virtio_serial_file_transfer_iommu.cfg

```diff
@@ -0,0 +1,66 @@
+- virtio_serial_file_transfer_iommu:
+    type = virtio_serial_file_transfer
+    guest_script_folder = /var/tmp/
+    file_transfer_serial_port = vs1
+    serials += " vs1"
+    serial_type_vs1 = virtserialport
+    file_sender = both
+    transfer_timeout = 720
+    clean_cmd = rm -f
+    filesize = 10
+    host_script = serial_host_send_receive.py
+    guest_script = VirtIoChannel_guest_send_receive.py
+    virtio_dev_iommu_platform = on
+    virtio_dev_disable_legacy = on
+    virtio_dev_disable_modern = off
```

Comment:
Only one concern, this cfg seems only 3 lines diff to `virtio_serial_file_transfer.cfg`, do we need a new file?
But if you guys all okay about it, I can merge it as well.

Reply:
Per you,"Serial can be launched without IOMMU configuration, so this would be a new case/variant." so creating a new file for it.  I think we can cover the two different type of test at the same time. Thanks.

Reply:
> Only one concern, this cfg seems only 3 lines diff to `virtio_serial_file_transfer.cfg`, do we need a new file?
> But if you guys all okay about it, I can merge it as well.

I'm ok with new cfg or new variants in old cfg, thanks : )

Reply:
Well, what's the difference for case `virtio_serial_file_transfer_iommu`  and `virtio_serial_file_transfer..iommu_enabled`  with x86 platform ? @nanliu-r 
ppc may do not support intel_iommu device  and do not need to enable guest iommu,  IMO this is configurable, one cfg is enough.

Reply:
> Well, what's the difference for case `virtio_serial_file_transfer_iommu` and `virtio_serial_file_transfer..iommu_enabled` with x86 platform ? @nanliu-r
> ppc may do not support intel_iommu device and do not need to enable guest iommu, IMO this is configurable, one cfg is enough.

@yanan-fu mdeng means that we want to cover two scenarios: 1 - with iommu, 2 - without iommu.
It's ok to use the old cfg to cover two senario by adding one new variants. 
Do I understand right? @MiriamDeng

Reply:
> > Well, what's the difference for case `virtio_serial_file_transfer_iommu` and `virtio_serial_file_transfer..iommu_enabled` with x86 platform ? @nanliu-r
> > ppc may do not support intel_iommu device and do not need to enable guest iommu, IMO this is configurable, one cfg is enough.
> 
> @yanan-fu mdeng means that we want to cover two scenarios: 1 - with iommu, 2 - without iommu.
> It's ok to use the old cfg to cover two senario by adding one new variants.
> Do I understand right? @MiriamDeng
Yes. Correct.

Reply:
@MiriamDeng  Without this patch, x86 (q35 machine type) already cover the two scenario: with iommu and without iommu.  This patch is totally duplicate for x86.

Update the existing cases `iommu_enabled` to cover ppc is ok.

Reply:
Suggest update case `iommu_enabled`:
```
        - iommu_enabled:                                                        
            virtio_dev_iommu_platform = on                                      
            x86_64, i386:                                                       
                only q35                                                        
                only unix_socket                                                
                no WinXP WinVista Win7 Win8 Win8.1 Win2000 Win2003              
                no Win2008 Win2008..r2 Win2012 Win2012..r2                      
                intel_iommu = yes                                               
                virtio_dev_iommu_platform = on                                  
                iommu_eim = on                                                  
                enable_guest_iommu = yes                                        
                virtio_dev_ats = on                                             
                machine_type_extra_params = "kernel-irqchip=split"              
            ppc64le,***:                                                        
                ppc_only_configuration
```

And for the ppc part, i think `virtio_dev_disable_legacy = on` and `virtio_dev_disable_modern = off` is not very meaningful, it has been the qemu default behavioru for long time, but i am ok to keep it by force, it is up to you.

Reply:
Please ignore the pre comment, i misunderstand the requirement.

Reply:
@yanan-fu ,it's not the case if we add a new cfg file since the test cases were generated by above code you provided are not same.

Reply:
@MiriamDeng  To cover all the cases with iommu, we need a new variant, how about :
```
    variants:                                                                      
        - @iommu_disabled:                                                              
        - iommu_enabled:                                                           
            virtio_dev_iommu_platform = on                                         
            x86_64, i386:                                                          
                only q35                                                           
                only unix_socket                                                   
                only default         --------> @nanliu-r Please check if x86 need other tests or not                                              
                no WinXP WinVista Win7 Win8 Win8.1 Win2000 Win2003                 
                no Win2008 Win2008..r2 Win2012 Win2012..r2                         
                intel_iommu = yes                                                  
                iommu_eim = on                                                     
                enable_guest_iommu = yes                                           
                virtio_dev_ats = on                                                
                machine_type_extra_params = "kernel-irqchip=split"                 
            ppc64****.***:                                                         
                ppc specified_configuration 
```

Reply:
By your way, it only generated several test cases as followings.
 (09/15) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.iommu_enabled_power:
 (15/15) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.virtio_serial_file_transfer.pty.iommu_enabled_power
Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.virtio_serial_file_transfer.tcp_socket.iommu_enabled_power

But with a new cfg file, all the test cases can be run with "virtio_dev_iommu_platform = on ...", it's the intention for me to send this patch, as you said it's duplicated for x86, do you think it's proper for me to mark the new file as ppc only ?

Reply:
It's reasonable for the second way, I will refer to it, thanks.